src/api/wallet/WalletApiMock.ts

```diff
@@ -44,7 +48,31 @@ export class WalletApiMock implements WalletApi {
   public async getNetworkId(): Promise<number> {
     assert(this._connected, 'The wallet is not connected')
 
-    return Network.Rinkeby
+    return this._networkId
+  }
+
+  public addOnChangeWalletInfo(callback: (walletInfo: WalletInfo) => void, trigger?: boolean): void {
+    this._listeners.push(callback)
+    if (trigger) {
+      callback(this._getWalletInfo())
+    }
```

Comment:
consider returning `removeOnChangeWalletInfo(callback)` for ease of unsubscribe, especially inside `useEffect`

Reply:
Not sure if you meant exactly [this](https://github.com/gnosis/dex-react/pull/83/commits/36053915fee69995bffd98e0c553a114f695ae33), but it was a good suggestion: